tests: obj_validation: enable HW_STACK_PROTECTION

tyrantlizard77 opened the pull request:
CONFIG_HW_STACK_PROTECTION is a dependency of CONFIG_USERSPACE.
Fixes a problem where userspace wasn't enabled even if the target HW
platform supports it.

Also fix a bug on ARM targets with gen_kobject_list.py

Commit:
tests: obj_validation: enable HW_STACK_PROTECTION

CONFIG_HW_STACK_PROTECTION is a dependency of CONFIG_USERSPACE.
Fixes a problem where userspace wasn't enabled even if the target HW
platform supports it.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

andrewboie commented:
I am wondering, is this the right fix?
Or should I modify arch/Kconfig to have CONFIG_USERSPACE select CONFIG_HW_STACK_PROTECTION instead of depending on it?

andrewboie commented:
this has exposed a bug with how gen_syscalls.py determines what subsystem drivers belong to. looking into it now.

Commit:
gen_kobject_list: fix on ARM targets

On ARM, a zero memory address actually falls within the expected bounds
of kernel memory.

Move the NULL check outside the bounds check, so that kernel objects
with NULL memory addresses in the DWARF info (because gc-sections
discarded them) won't confound the script's logic.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

andrewboie commented:
going to take the alternate approach
